fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -90,4 +91,15 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Decides how "enchant-able" this item is, for example enchanting a gold tool will net you more enchants than
+	 * enchanting a diamond one. Stack-aware version of {@link Item#getEnchantability()}.
+	 *
+	 * @see ToolMaterials#getEnchantability()
+	 * @param stack the current stack
+	 */
+	default int getEnchantability(ItemStack stack) {
+		return ((Item) this).getEnchantability();
```

Comment:
Maybe is useful to inject this method in `ItemStack` with a `FabricItemStack` interface?

```java
public interface FabricItemStack {
	default int getEnchantability() {
		return ((ItemStack) this).getItem().getEnchantability((ItemStack) this);
	}
}
```

Reply:
Not sure, I don't think many people will be calling this method?

Reply:
I mean for completion, this is similar to `getRecipeRemainder()` of stack aware remainders.
However is better to do this in a future pr, after the remainders one is merged, so FabricItemStack already exists.